workflow/executor/k8sapi/client.go

```diff
@@ -116,7 +116,7 @@ func (c *k8sAPIClient) until(ctx context.Context, f func(pod *corev1.Pod) bool)
 		done, err := func() (bool, error) {
 			w, err := podInterface.Watch(ctx, metav1.ListOptions{FieldSelector: "metadata.name=" + c.podName})
 			if err != nil {
-				return false, err
+				return true, fmt.Errorf("failed to establish pod watch: %w", err)
```

Comment:
This is where the error occurs. However, the error message isn't that helpful:
```
time="2021-03-12T04:28:16.806Z" level=error msg="executor error: failed to wait for main container to complete: timed out waiting for the condition: failed to establish pod watch: unknown (get pods)"
```

which is why I decided to print the response body of the actual request in `k8sLogRoundTripper` so we know what exactly went wrong.

Reply:
@alexec WDYT? Would you like me to remove the change in k8sLogRoundTripper so this PR can focus on resolving the stuck wait container issue? 

Is there any concern on printing out the response body in k8sLogRoundTripper? The code is more complex but I think it’s pretty helpful for debugging sometimes.